node/narwhal/tests/bft_e2e.rs

```diff
@@ -123,3 +124,50 @@ async fn test_quorum_break() {
     // Check the nodes have stopped advancing through the rounds.
     assert!(network.is_halted().await);
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_leader_election_consistency() {
+    // The round at which to start checking for leader consistency.
+    const STARTING_ROUND: u64 = 2;
+
+    // The minimum and maximum rounds to check for leader consistency.
+    const MIN_ROUND: u64 = 4;
```

Comment:
sooo is this one or `STARTING_ROUND` the one to start checking for leader consistency :thinking:? this should be clarified